src/Select.tsx

```diff
@@ -340,9 +340,9 @@ export const Select: FC<Partial<SelectProps>> = ({
       }
 
       if (index > -1 || createable) {
-        const newSelection = result[index]?.value || {
-          value: inputValue,
-          children: inputValue
+        const newSelection = {
```

Comment:
Wow - this was it? Was it just passing the wrong var?

Reply:
yup, I verified through storybook :)